portal/migrations/versions/0701782c564d_.py

```diff
@@ -0,0 +1,112 @@
+"""Correct user_consents on deceased patients
+
+Revision ID: 0701782c564d
+Revises: 1d84237ed07c
+Create Date: 2019-03-28 12:16:06.567817
+
+"""
+import logging
+
+from portal.database import db
+from portal.models.audit import Audit
+from portal.models.qb_timeline import invalidate_users_QBT
+from portal.models.user_consent import (
+    STAFF_EDITABLE_MASK,
+    INCLUDE_IN_REPORTS_MASK,
+    SEND_REMINDERS_MASK,
+    UserConsent,
+)
+from portal.models.user import User
+
+# revision identifiers, used by Alembic.
+revision = '0701782c564d'
+down_revision = '1d84237ed07c'
+
+log = logging.getLogger("alembic")
+log.setLevel(logging.DEBUG)
+
+
+def patch_deceased_user_consents(uid):
+    user = User.query.get(uid)
+    user_consents = UserConsent.query.filter(
+        UserConsent.user_id == uid).order_by(UserConsent.id).all()
+
+    # if no user_consents exist for user, we're done
+    if not user_consents:
+        return
+
+    # acquire the deceased date
+    dd = user.deceased.timestamp
+
+    # if the most recent consent looks correct, assume we're done
+    if user_consents[-1].status == 'suspended' and not(
+            user_consents[-1].options & SEND_REMINDERS_MASK):
+        return
+
+    admin = User.query.filter_by(email='__system__').first()
+
+    log.info("Patching broken consents for {}, with deceased date {}".format(
+        user, dd))
+    keeper = user_consents.pop(0)
+    log.info(" keeping original user_consent {} {} {} {}".format(
```

Comment:
Probably not worth it now, but if you end up having to correct these a lot, might be worth adding a `__repr__` method to `UserConsent`